### PR TITLE
Add Schema and code for storing daily view on user profiles

### PR DESCRIPTION
**Pull request by shyam0705**
## Changes proposed
Added Schema and code for storing daily views on user profiles and added separate section for showing the most popular profiles of the last 28 days.

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Note to reviewers

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/2691"></a>

### CODE REVIEW

File: models/ProfileStats.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  },
+});
+
+const profileStatsSchema = new mongoose.Schema({
+  username: {
+    type: String,
+    required: true,
+  },
+  views: {
+    type: [viewSchema],
```
Comment (eddiejaoude) [inline models/ProfileStats.js:19]: I think simplifying the model will make our code cleaner, similar to the way links are done https://github.com/EddieHubCommunity/LinkFree/blob/main/models/Link.js

Comment (shyam0705) [reply models/ProfileStats.js:19]: do you suggesting schema like { date, profile views} ?

Comment (eddiejaoude) [reply models/ProfileStats.js:19]: Yes I think so. Here it is formatted for others to be able to read...

```js
new mongoose.Schema({
   username: String,
   views: {
      type: Number,
      default: 0,
   },
   date:{
      type: Date,
      default: new Date(new Date().toLocaleDateString()),
   }
});
```

It is probably worth creating a relationship to the profile with

```js
   profile: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "profile",
    },
```

Comment (shyam0705) [reply models/ProfileStats.js:19]: Thanks for the suggestion, @eddiejaoude. Working on it

**Comment by shyam0705** [comment]
@eddiejaoude I have simplified the schema and added a section showing popular profiles in last 28 days, and the project is running fine locally.

UI:-

image

Database:-

image

I don't know why one test is failing here when I tested locally, that test is passing correctly.

image

**Comment by eddiejaoude** [comment]
I think this is a flakey test, let me take a look

**Review by eddiejaoude — APPROVED** [review]
Looks good 👍 

I will merge into a temporary branch, and combine on the new page `Discover` that replaces the `Popular` profiles page

**Pull request merged**